### DISCUSSION TITLE
question about fltk capabilities

### DISCUSSION THREAD

**Discussion by DavidBJaffe** [fltk-rs/fltk-rs#690] [Q&A]
I work on a fairly large Rust program and need to add GUI code to it for use on a Mac.
I would be grateful for input as to whether this could work using the `fltk` crate.  Here are the needs:

1. The existing code makes SVG objects with tooltip information embedded within.  We need the
objects to be displayed and the tooltip information to show up when the corresponding entity is
hovered over.  We can restructure the tooltip information or move it out of the SVG objects if needed.

2. We have tables that are represented as UTF-8 strings, and which include ANSI escape codes for
coloring and bolding, and which are in a particular font that works properly for this purpose.  We need
to be able to display these.  We have HTML versions if that would help.

3. The SVG objects and tables will appear in a window, with the objects organized vertically, which the 
user can scroll through.

**Comment by MoAlyousef** [comment]
Hi

1- You can create SvgImage objects and display them in frames, you can either use the set_tooltip method for those frames, or you can handle the Event::Enter & Event::Leave to support hovering, and display a custom window with those tooltips if needed.

2- FLTK has tables, the api is a bit complex, they support UTF-8 strings, but doesn't support ansi escape sequences. So you would have to manually modify the data text in each cell depending on the original string. You would need to use FLTK's drawing primitives to modify the properties/color/font of the string. It might require a lot of boilerplate.

3- There's a Scroll widget which support scrolling.
